embedly/embedly/extract.py

```diff
@@ -23,7 +27,12 @@ def _get_cached_url(self, url):
         cached_data = self.redis_client.get(cache_key)
 
         if cached_data is not None:
-            return json.loads(cached_data)
+            try:
+                return json.loads(cached_data)
+            except ValueError:
+                raise URLExtractorException(
```

Comment:
Which raises a larger question which is, if a single portion of the request fails, ie failure to read from the cache or failure to query embedly, should we return a partially satisfied query for the URLs whose data we are able to query?  In which case do we signal to the client that we were only able to find the metadata for some of the URLs?